design/types.go

```diff
@@ -133,6 +133,8 @@ const (
 	UserTypeKind
 	// MediaTypeKind represents a media type.
 	MediaTypeKind
+	// NoneKind represents a nil
```

Comment:
I think I am leaning toward `NoExample()` in the end. The `NoneKind` seems a bit too much for one purpose I agree.